2-1.cpp

```diff
@@ -0,0 +1,92 @@
+﻿#include <iostream>
+using namespace std;
+/*
+* \brief увеличение числа a в n раз
+* \param a число a
+* \param n число n
+* \return число a увеличенное в n раз
+*/
+int getproduct(int a, int n);
+
+/*
+* \brief увеличение числа a в 2n раз
+* \param a число a
+* \param n число n
+* \return число a увеличенное в 2n раз
+*/
+int getdoubleproduct(int a, int n);
+
+/*
+* \brief увеличение числа a в 3n раз
+* \param a число a
+* \param n число n
+* \return число a увеличенное в 3n раз
+*/
+int gettripleproduct(int a, int n);
+
+/*
+* \brief выбор заполнения массива
+*/
+enum class path
+{
+	Single = 1,
+	Double = 2,
+	Triple = 3
+};
+
+/*
+* \brief Точка входа в программу.
+* \return Код ошибки, если 0 - успешное выполнение.
+*/
+int main()
+{
+	setlocale(LC_ALL, "Russian");
+	cout << "Введите число A : ";
+	int a;
+	cin >> a;
+	cout << "Введите число N : ";
+	int n;
+	cin >> n;
+	cout << "Выберите формулу:\n1: A * N\n2: 2 * A * N\n3: 3 * A * N\nВыбор: ";
+	int choice;
+	cin >> choice;
+	const auto filling = static_cast<path>(choice);
+	switch (filling) {
+	case path::Single:
+		{
+			cout << "A * N = " << getproduct(a, n);
+			break;
+		}
+	case path::Double:
+		{
+			cout << "2 * A * N = " << getdoubleproduct(a, n);
+			break;
+		}
+	case path::Triple:
+		{
+			cout << "3 * A * N = " << gettripleproduct(a, n);
+			break;
+		}
+	default:
+	{
+		cout << "Некоректный ввод.";
+		return (0);
+	}
+	}
+	return 0;
+}
+
+int getproduct(int a, int n)
```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.